qiskit/visualization/latex.py

```diff
@@ -345,690 +321,303 @@ def _get_beamer_page(self):
+                    nxt -= 1
+            else:
+                nxt += 1
+                while nxt not in wire_list:
+                    nxt += 1
+            if cond == '0':
+                self._latex[pos][col] = "\\ctrlo{" + str(
+                    nxt - wire_list[index]) + "}"
+            elif cond == '1':
+                self._latex[pos][col] = "\\ctrl{" + str(
+                    nxt - wire_list[index]) + "}"
+
+    def _add_params_to_gate_text(self, op, gate_text):
+        """Add the params to the end of the current gate_text"""
+        # Must limit to 4 params or may get dimension too large error
+        # from xy-pic xymatrix command
+        if (len(op.op.params) > 0 and not any(
+                isinstance(param, np.ndarray) for param in op.op.params)):
+            gate_text += "\\,\\mathrm{(}"
+            if len(op.op.params) > 4:
+                gate_text += '\\mathrm{>4\\,parameters}'
+            else:
+                for param in op.op.params:
+                    gate_text += "\\mathrm{%s}," % pi_check(param, output='latex', ndigits=4)
+                gate_text = gate_text[:-1]
+            gate_text += "\\mathrm{)}"
+        return gate_text
+
+    def _add_condition(self, op, wire_list, col):
+        """Add a condition to the _latex list"""
+        # if_value - a bit string for the condition
+        # cwire - the wire number for the first wire for the condition register
+        #         or if cregbundle, wire number of the condition register itself
+        # gap - the number of wires from cwire to the bottom gate qubit
+
+        creg_size = self.cregs[op.condition[0]]
+        if_value = format(op.condition[1], 'b').zfill(creg_size)
+        if not self.reverse_bits:
+            if_value = if_value[::-1]
+
+        cwire = len(self.qubit_list)
+        iter_cregs = iter(list(self.cregs)) if self.cregbundle else iter(self.cregs_bits)
+        for creg in iter_cregs:
+            if creg == op.condition[0]:
+                break
+            cwire += 1
 
-    Args:
-        matchobj (re.Match): contains original float
-        ndigits (int): Number of digits to print
-    Returns:
-       str: returns truncated float
-    """
-    if matchobj.group(0):
-        return '%.{}g'.format(ndigits) % float(matchobj.group(0))
-    return ''
+        gap = cwire - max(wire_list)
+        if self.cregbundle:
+            # Print the condition value at the bottom
+            self._latex[cwire][col] = \
+                "\\dstick{_{_{=%s}}} \\cw \\cwx[-%s]" % (str(op.condition[1]), str(gap))
+        else:
+            for i in range(creg_size):
+                if if_value[i] == '1':
+                    self._latex[cwire + i][col] = \
+                        "\\control \\cw \\cwx[-" + str(gap) + "]"
+                    gap = 1
+                else:
+                    self._latex[cwire + i][col] = \
+                        "\\controlo \\cw \\cwx[-" + str(gap) + "]"
+                    gap = 1
```

Comment:
```suggestion
                if if_value[i] == '1':
                    control = '\\control'
                else:
                    control = '\\controlo'
                    
              self._latex[cwire + i][col] = f"{control} \\cw \\cwx[-" + str(gap) + "]"
              gap = 1
```

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/5855/commits/2ff9150c1a0e7d6ccf38280ca16f76696d10697e.